src/app/controllers/charting.controller.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * ngInject
+ */
+export function ChartingController($scope, $attrs, $element, Utils) {
+  let defaultMargins = {left: 30, top: 10, bottom: 30, right: 10};
+  $scope.config = {};
```

Comment:
Anything attached to scope here, should really be attached to the controller instance instead.

The link function in the directive takes a fourth argument, which is the controller instance. So you can directly access controller methods and properties there if necessary. $scope should be kept as clean as possible.